geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/data/stats/GeoMesaMetadataStats.scala

```diff
@@ -212,6 +224,13 @@ class GeoMesaMetadataStats(val ds: AccumuloDataStore, statsTable: String, genera
 
   override def clearStats(sft: SimpleFeatureType): Unit = metadata.delete(sft.getTypeName)
 
+  override def close(): Unit = {
+    running.set(false)
+    if (scheduledCompaction != null) {
+      scheduledCompaction.cancel(false)
+    }
```

Comment:
yeah, I decided to not worry about synchronizing it because the compactions are only run once an hour, so the odds of an NPE are low (and the data store would have to be held for at least an hour before disposing)... but true it could throw during dispose, which would be unfortunate. I'll update